cpp/src/arrow/flight/sql/odbc/flight_sql/flight_sql_statement_get_columns.cc

```diff
@@ -115,7 +115,10 @@ Result<std::shared_ptr<RecordBatch>> Transform_inner(
       odbcabstraction::SqlDataType data_type_v3 =
           GetDataTypeFromArrowField_V3(field, metadata_settings.use_wide_char_);
 
-      ColumnMetadata metadata(field->metadata());
+      const auto& metadata_map = field->metadata();
+      std::shared_ptr<const arrow::KeyValueMetadata> empty_metadata_map(
+          new arrow::KeyValueMetadata);
+      ColumnMetadata metadata(metadata_map ? metadata_map : empty_metadata_map);
```

Comment:
Let's change the ColumnMetadata object itself to handle NULL safely and internally create an empty owned map (since it's easy for this to occur elsewhere)

Reply:
yea that makes sense, I have updated the code to change `ColumnMetadata`. 

There is an API `Field::HasMetadata` in arrow that checks if metadata is null, but seems it's not as commonly used.
https://github.com/Bit-Quill/arrow/blob/a6bc579a7bca64901e7a407e995c91cbc8816519/cpp/src/arrow/type.cc#L304-L306